test/python/quantum_info/operators/symplectic/test_pauli_list.py

```diff
@@ -0,0 +1,1470 @@
+            target = PauliList(["IIII", "YYYY", "XIIY", "YIIZ"])
+            self.assertEqual(pauli1 + pauli2([0, 3]), target)
+
+        with self.subTest(msg="qargs=[2, 1]"):
+            target = PauliList(["IIII", "YYYY", "IYXI", "IZYI"])
+            self.assertEqual(pauli1 + pauli2([2, 1]), target)
+
+        with self.subTest(msg="qargs=[3, 1]"):
+            target = PauliList(["IIII", "YYYY", "YIXI", "ZIYI"])
+            self.assertEqual(pauli1 + pauli2([3, 1]), target)
+
+    def test_getitem_methods(self):
+        """Test __getitem__ method."""
+        with self.subTest(msg="__getitem__ single"):
+            labels = ["XI", "IY"]
+            pauli = PauliList(labels)
+            self.assertEqual(pauli[0], PauliList(labels[0]))
+            self.assertEqual(pauli[1], PauliList(labels[1]))
+
+        with self.subTest(msg="__getitem__ array"):
+            labels = np.array(["XI", "IY", "IZ", "XY", "ZX"])
+            pauli = PauliList(labels)
+            inds = [0, 3]
+            self.assertEqual(pauli[inds], PauliList(labels[inds]))
+            inds = np.array([4, 1])
+            self.assertEqual(pauli[inds], PauliList(labels[inds]))
+
+        with self.subTest(msg="__getitem__ slice"):
+            labels = np.array(["XI", "IY", "IZ", "XY", "ZX"])
+            pauli = PauliList(labels)
+            self.assertEqual(pauli[:], pauli)
+            self.assertEqual(pauli[1:3], PauliList(labels[1:3]))
+
+    def test_setitem_methods(self):
+        """Test __setitem__ method."""
+        with self.subTest(msg="__setitem__ single"):
+            labels = ["XI", "IY"]
+            pauli = PauliList(["XI", "IY"])
+            pauli[0] = "II"
+            self.assertEqual(pauli[0], PauliList("II"))
+            pauli[1] = "XX"
+            self.assertEqual(pauli[1], PauliList("XX"))
+
+            with self.assertRaises(Exception):
+                # Wrong size Pauli
+                pauli[0] = "XXX"
+
+        with self.subTest(msg="__setitem__ array"):
+            labels = np.array(["XI", "IY", "IZ"])
+            pauli = PauliList(labels)
+            target = PauliList(["II", "ZZ"])
+            inds = [2, 0]
+            pauli[inds] = target
+            self.assertEqual(pauli[inds], target)
+
+            with self.assertRaises(Exception):
+                pauli[inds] = PauliList(["YY", "ZZ", "XX"])
+
+        with self.subTest(msg="__setitem__ slice"):
+            labels = np.array(5 * ["III"])
+            pauli = PauliList(labels)
+            target = PauliList(5 * ["XXX"])
+            pauli[:] = target
+            self.assertEqual(pauli[:], target)
+            target = PauliList(2 * ["ZZZ"])
+            pauli[1:3] = target
+            self.assertEqual(pauli[1:3], target)
+
+
+class TestPauliListLabels(QiskitTestCase):
+    """Tests PauliList label representation conversions."""
+
+    def test_from_labels_1q(self):
+        """Test 1-qubit from_labels method."""
+        labels = ["I", "Z", "Z", "X", "Y"]
```

Comment:
Z appears twice

Reply:
This is as intended. We also see that the same Pauli is not omitted.